src/server/qgswfsserver.cpp

```diff
@@ -2150,6 +2161,12 @@ QDomElement QgsWFSServer::createFeatureGML3( QgsFeature* feat, QDomDocument& doc
   featureElement.appendChild( typeNameElement );
 
   const QgsGeometry* geom = feat->constGeometry();
+  if ( isMultiGeom && QgsWKBTypes::isSingleType( QGis::fromOldWkbType( geom->wkbType() ) ) )
+  {
+    QgsGeometry* cloneGeom = new QgsGeometry( *geom );
```

Comment:
Memory leak.

Reply:
Same here -- this should not be a pointer in the first place.